src/bulk/common.ts

```diff
@@ -579,23 +579,18 @@ function executeCommands(
   }
 
   try {
-    if (isInsertBatch(batch)) {
-      executeOperation(
-        bulkOperation.s.collection.client,
-        new InsertOperation(bulkOperation.s.namespace, batch.operations, finalOptions),
-        resultHandler
-      );
-    } else if (isUpdateBatch(batch)) {
-      executeOperation(
-        bulkOperation.s.collection.client,
-        new UpdateOperation(bulkOperation.s.namespace, batch.operations, finalOptions),
-        resultHandler
-      );
-    } else if (isDeleteBatch(batch)) {
-      executeOperation(
-        bulkOperation.s.collection.client,
-        new DeleteOperation(bulkOperation.s.namespace, batch.operations, finalOptions),
-        resultHandler
+    const operation = isInsertBatch(batch)
+      ? new InsertOperation(bulkOperation.s.namespace, batch.operations, finalOptions)
+      : isUpdateBatch(batch)
+      ? new UpdateOperation(bulkOperation.s.namespace, batch.operations, finalOptions)
+      : isDeleteBatch(batch)
+      ? new DeleteOperation(bulkOperation.s.namespace, batch.operations, finalOptions)
+      : null;
```

Comment:
It's not a case we currently handle - but what happens if we can't find a batch?  In theory that's an impossible state.  Should we throw a `MongoRuntimeError` if we are unable to create an operation to execute?

Reply:
I considered that but I decided against making functional changes. I agree it is an impossible state but do we know for sure that the "empty" code path is unused? if it is I'm not sure anything would crash, I can make a ticket to follow up on it.

Reply:
I think it's a relatively safe assumption - `batch` is defined above as:

```typescript
if (bulkOperation.s.batches.length === 0) {
    return callback(
      undefined,
      new BulkWriteResult(bulkOperation.s.bulkResult, bulkOperation.isOrdered)
    );
  }
  const batch = bulkOperation.s.batches.shift() as Batch;
```

so batch will never be empty

Reply:
https://jira.mongodb.org/browse/NODE-5813

Reply:
It looks like addToOperationsList does not validate the BatchType, so it could be input wrong (even though TS would error on it) and then at runtime this logic would silently skip. I don't think that is correct but it is worthy of a thorough look before it changes.